pkg/api/controller.go

```diff
@@ -2196,7 +2196,7 @@ func (c *Controller) GetCommit(w http.ResponseWriter, r *http.Request, repositor
 		return
 	}
 	ctx := r.Context()
-	c.LogAction(ctx, "get_commit")
+	c.LogAction(ctx, "get_commit", r.UserAgent(), repository, commitID, "")
```

Comment:
AFAIC it's not at the moment. 
Might be in the future - if we collect commit id in other actions like create commit and not just at the get commit point.

Reply:
@itaiad200 it is also two for other operations - diff, merge and create branch.

Reply:
@nopcoder create branch only gets 'branch' as ref according to the table you added on the issue associated #454

@itaiad200 I agree about the commit ID at this point, it's not part of the requirements, let's consider not using it in this constellation.
I share your concern about multiple values with different meaning for this field (ref/branch/commit ID).
Going back to the original requirement - we wanted to collect branch, and then came up the question - what branch do we keep for events like 'merge'? we do need both source and destination. (similar case with diff)
This is why came up the idea of additional field 'source ref', but now I'm not so sure keeping multiple fields for these unique cases seems reasonable.

I don't want to block this PR, I think we should stick to the original plan and add branch field with branch values **only** at this point, going forward if we see fit we can add the others in some form.

As for the 'merge' events I'm thinking maybe we can keep the **source branch** as 'branch' value and create another table for merges with a foreign key field, on the 'events' table side, referring to 'merge'(installation id, time, source, destination) table. On the 'merge' table we can keep any other information about the merge if there is. 
Makes sense? WDYT?

Reply:
I see many BI collections that use the `ref` as is. It's more than the `CommitID` being probably redundant and will hard to deduct any meaningful insights from it. It also masks important branch information. However I understand the limitations - `Ref` interpolation is happening in Catalog/Graveler, and not in the Controller where BI is collected. I'm fine with collecting it as it's in this PR, as long as @NoyDavidson2 is onboard with collecting `Refs` which is a fancy word for either Branch/Commit/Traverse of the commit tree (`main^2~3`)

Reply:
1. Got the linked issue wrong. The number is from a different repo - will probably need to replace it with non link or a different issue.
2. There is no point in linking another table with extra properties for each event, unless we create different tables for different types as the relation is 1:1 + it will require adding unique ID to the event to bind the extra information.
3. Wanted to make it clear, as @itaiad200 said - almost all places that we accept 'ref' and not 'branch' can report commit id (event short), branch, tag or relative commit which will make the different IDs to work with.
If we plan to use just branch names - at the point we report the metric we can limit the branch name just to operations that accepts branch name. It in case of 'merge' the target branch name. But in case of 'diff' we don't have any promise we compare two branches.

Reply:
Update - @nopcoder and I have spoke this morning and decided to release this feature the way it is described here. After working with the data we will decide if and how we can make it better as we go.